setup.py

```diff
@@ -41,6 +44,41 @@
     ],
 }
 
+
+def _executable_exists(executable_name):
+    """Check for existence of executable in environment.
+    """
+    exec_check_cmd = ['/usr/bin/which', executable_name]
+    try:
+        subprocess.check_call(exec_check_cmd)
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+class BuildPyCommand(setuptools.command.build_py.build_py):
+    """Custom build command for packaging."""
+
+    def _make_thrift(self):
+        """Generate Baseplate Thrift definitions.
+        """
+        # Verify build dependencies are installed.
+        if not _executable_exists('make'):
+            print("'make' not found.")
+            exit(1)
+
+        if not _executable_exists('thrift1'):
+            print("Thrift compiler not found.")
+            exit(1)
+
+        make_cmd = ['/usr/bin/make', 'thrift']
+        subprocess.check_call(make_cmd)
+
+    def run(self):
+        self._make_thrift()
+        setuptools.command.build_py.build_py.run(self)
```

Comment:
Does this not support `super`?

Reply:
This was the example I followed from https://seasonofcode.com/posts/how-to-add-custom-build-steps-and-commands-to-setuppy.html (the very end). 

I don't believe this particular class supports super -> https://github.com/pypa/setuptools/blob/master/setuptools/command/build_py.py#L23 + https://github.com/python/cpython/blob/2.7/Lib/distutils/command/build_py.py#L16 

These don't appear to be `object`-based python classes.

Reply:
Not surprised :)